tests/test262_runner.rs

```diff
@@ -0,0 +1,589 @@
+                && Self::filter_file(filters, &entry.file_name())
+            {
+                self.run_test(&entry.path());
+            }
+        }
+    }
+
+    fn filter_dir(filters: &[PathBuf], os_file_name: &OsStr) -> Option<Box<[PathBuf]>> {
+        if SKIP_DIRS.contains(&os_file_name.to_str().unwrap()) {
+            return None;
+        }
+        if filters.is_empty() {
+            return Some(vec![].into_boxed_slice());
+        }
+
+        let mut child_filters = Vec::with_capacity(filters.len());
+        for filter in filters {
+            if filter == OsStr::new("") || filter == os_file_name {
+                return Some(vec![].into_boxed_slice());
+            }
+            if let Ok(child_filter) = filter.strip_prefix(os_file_name) {
+                child_filters.push(child_filter.to_path_buf());
+            }
+        }
+
+        if child_filters.is_empty() {
+            None
+        } else {
+            Some(child_filters.into_boxed_slice())
+        }
+    }
+
+    fn filter_file(filters: &[PathBuf], os_file_name: &OsStr) -> bool {
+        if !is_test_file(os_file_name.to_str().unwrap()) {
+            return false;
+        }
+        if filters.is_empty() {
+            return true;
+        }
+        filters.iter().any(|filter| filter == os_file_name)
+    }
+
+    fn run_test(&mut self, path: &PathBuf) {
+        //println!("Running test {:?}", path);
+        let metadata = test_metadata::parse(path);
+
+        if metadata.flags.is_async || metadata.flags.module {
+            // We don't yet support async or modules, skip any tests for them.
+            return;
+        }
+
+        self.num_tests_run += 1;
+
+        let mut command = Command::new(&self.nova_cli_path);
+        command.arg("eval");
+
+        command.arg(&self.nova_harness_path);
+        if metadata.flags.raw {
+            assert!(metadata.includes.is_empty());
+        } else {
+            let mut harness = self.runner_base_path.clone();
+            harness.push("test262/harness");
+            let includes_iter = ["assert.js", "sta.js"]
+                .iter()
+                .map(PathBuf::from)
+                .chain(metadata.includes);
+            for include_relpath in includes_iter {
+                let mut include = harness.clone();
+                include.push(include_relpath);
+                command.arg(include);
+            }
+        }
+        command.arg(path);
+
+        //println!("{:?}", command);
+
+        let test_result = Self::handle_test_output(command, &metadata.negative);
+
+        let relpath = path.strip_prefix(&self.tests_base).unwrap();
+        let expectation = self
+            .expectations
+            .get(relpath)
+            .copied()
+            .unwrap_or(TestExpectation::Pass);
+
+        if test_result != expectation {
+            if self.treat_crashes_as_failures
+                && matches!(test_result, TestExpectation::Fail | TestExpectation::Crash)
```

Comment:
**suggestion:** This if case could also use a comment to explain what we're doing here. (Oh, I think I got it now as I was writing this comment.)